### PR TITLE
payment select modal

### PR DESCRIPTION
**Pull request by SamueleA**
Payment Select Modal V0. New UI will be added in another PR.

image

### CODE REVIEW

File: examples/react/src/components/Connected.tsx [hunk 1]
```diff
@@ -242,6 +248,31 @@ export const Connected = () => {
...
+      chainId,
+      priceRaw,
+      salesContractAddress,
+      recipientAddress: address,
+      currencyAddress,
+      nftId: tokenId,
+      nftAddress,
+      nftQuantity,
+      isDev: true,
+    }))
```
Comment (taylanpince) [inline examples/react/src/components/Connected.tsx:273]: @SamueleA This is really nice, I wonder if we can remove the `getSalesContractConfig` completely from this and simplify it further?

A few other notes:

- I think `tokenId` is a better name than `nftId`
- `collectionAddress` is better than `nftAddress`
- A way to control shown payment options would be nice (optionally)
- We should have Onramp and Transfer as options in there too
- `priceRaw` would be better as `tokenPrice` or just `price`
- Can we use the `network` type that accepts either chain ID or name?

Comment (SamueleA) [reply examples/react/src/components/Connected.tsx]: The requested changes have been addressed in subsequent commits!

File: packages/checkout/src/views/PaymentSelection/PayWithCreditCard.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      flexDirection={isMobile ? 'column' : 'row'}
+      alignItems="center"
+      justifyContent="space-between"
+      gap={isMobile ? '2' : '0'}
+      style={{
+        minHeight: getCardHeight(isMobile)
+      }}
+    >
+      <Box justifyContent={isMobile ? 'center' : 'flex-start'}>
+        <Text color="text100">Buy With Credit Card</Text>
```
[the remark this answers is not in the document]
Comment (SamueleA) [reply packages/checkout/src/views/PaymentSelection/PayWithCreditCard.tsx]: Fixed in most recent commit!

File: packages/checkout/src/views/PaymentSelection/FiatOnRamp.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      flexDirection={isMobile ? 'column' : 'row'}
+      alignItems="center"
+      justifyContent="space-between"
+      gap={isMobile ? '2' : '0'}
+      style={{
+        minHeight: getCardHeight(isMobile)
+      }}
+    >
+      <Box justifyContent={isMobile ? 'center' : 'flex-start'}>
+        <Text color="text100">Purchase Crypto</Text>
```
[the remark this answers is not in the document]
Comment (SamueleA) [reply packages/checkout/src/views/PaymentSelection/FiatOnRamp.tsx]: Fixed in most recent commit!

File: packages/checkout/src/views/PaymentSelection/PayWithCrypto/PayWithMainCurrency.tsx [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+        minHeight: getCardHeight(isMobile)
+      }}
+    >
+      <Box
+        flexDirection="column"
+        gap="2"
+        justifyContent={isMobile ? 'center' : 'flex-start'}
+      >
+        <Box justifyContent={isMobile ? 'center' : 'flex-start'}>
+          <Text color="text100">Buy With {currencyInfoData?.name}</Text>
```
[the remark this answers is not in the document]
Comment (SamueleA) [reply packages/checkout/src/views/PaymentSelection/PayWithCrypto/PayWithMainCurrency.tsx]: Fixed in most recent commit!

File: packages/checkout/src/views/PaymentSelection/PayWithCrypto/SwapAndPay.tsx [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+          minHeight: getCardHeight(isMobile)
+        }}
+      >
+        <Box
+          flexDirection="column"
+          gap="2"
+          justifyContent={isMobile ? 'center' : 'flex-start'}
+        >
+          <Box justifyContent={isMobile ? 'center' : 'flex-start'}>
+            <Text color="text100">Buy With {swapQuote.info?.name || 'Unknown'}</Text>
```
[the remark this answers is not in the document]
Comment (SamueleA) [reply packages/checkout/src/views/PaymentSelection/PayWithCrypto/SwapAndPay.tsx]: Fixed in most recent commit!

**Review by tolgahan-arikan — APPROVED** [review]

**Review by corbanbrook — APPROVED** [review]

**Pull request merged**